mongodb/devhub
# DEVHUB-175: Automatically Add `tck` Param to Atlas Signup Urls

[JIRA](https://jira.mongodb.org/browse/DEVHUB-175)
[Staging Link (Relevant article)](https://docs-mongodbcom-staging.corp.mongodb.com/master/devhub/jordanstapinski/js/cloud-attributions/how-to/search-engine-using-atlas-full-text-search)
[Other Staging Article](https://docs-mongodbcom-staging.corp.mongodb.com/master/devhub/jordanstapinski/js/cloud-attributions/article/build-newsletter-website-mongodb-data-platform)

This PR adds some build-time logic to add an attribution `tck` query param to a specific URL supplied in the ticket `https://www.mongodb.com/cloud/atlas/signup`. There are a couple interesting cases that can come up:
- The url might already have some query params on it, so we want to be sure to preserve those
- The url might not have query params yet, so we want to be sure to add an `?` where needed
- We want to be sure to `encodeURIComponents`

~The actual `tck` value might change, but the logic is where we want it to be.~
The `tck` value is the last piece of the URL slug (aka the article title (but not always the title, we just use whatever the end of the slug is))

To Verify:
- Check out the two article staging links
- If you open the inspector console and cmd+f for signup, you should see the `tck` query param
- If you click these links the tck should stay, as well as any other query params
